src/modules/users/services/user.service.abstract.ts

```diff
@@ -0,0 +1,11 @@
+import type { ReturnUserDto, UpdateUserDto } from '../dtos';
+import type { User } from '../entities';
+
+interface IUserService {
```

Comment:
you can use it as an interface but in this case, you also need a symbol so you can use it with DI. Another way, you can use the abstract classes and call it as UserService and do implementation which you call as UserServiceImpl

Reply:
I've done the second implementation via abstract classes to pass it to 'provide' property of provider instead of Symbol of interface